src/Microsoft.Identity.Client/ApiEvent.cs

```diff
@@ -0,0 +1,48 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+
+namespace Microsoft.Identity.Client
+{
+    internal class ApiEvent : Event
+    {
+        public ApiEvent(string authority, string authorityType, string uiBehavior, string validationStatus, bool loginHint = false) : base("Microsoft.MSAL.api_event")
+        {
+            this["authority"] = authority;
+            this["authority_type"] = authorityType;
+            this["ui_behavior"] = uiBehavior;
+            this["validation_status"] = validationStatus;
+            this["login_hint"] = loginHint.ToString();
+        }
+
+        public void Update(string idToken, bool wasSuccessful)
```

Comment:
OK. Next commit will switch to normal setters.

The current implementation chose to use such helper function to group a batch of properties which are supposed to be set TOGETHER at the same moment. The individual setters do not have such group relationship. To that end, we may or may not still define some helpers for the grouping.